ompi/mca/coll/accelerator/coll_accelerator_reduce.c

```diff
@@ -84,3 +91,13 @@ mca_coll_accelerator_reduce(const void *sbuf, void *rbuf, size_t count,
     }
     return rc;
 }
+
+int
+mca_coll_accelerator_reduce_local(const void *sbuf, void *rbuf, size_t count,
+                                  struct ompi_datatype_t *dtype,
+                                  struct ompi_op_t *op,
+                                  mca_coll_base_module_t *module)
+{
+    return mca_coll_accelerator_reduce(sbuf, rbuf, count, dtype, op, -1, NULL,
```

Comment:
Instead of the solution proposed here, copy the code from above in this function. Cleaner and more readable from my perspective, as well as more into the OMPI approach to collective modules.

Reply:
@Akshay-Venkatesh Ping.

Reply:
@jsquyres / @bosilca Sorry this PR got de-prioritized last week. I've duplicated the code in the main reduce function now. Let me know if the changes look good.